compiler/lib/src/main/scala/codegen/CppWriter/FormatCppWriter.scala

```diff
@@ -56,11 +56,23 @@ object FormatCppWriter {
   }
 
   def write(f: Format, tn: AstNode[Ast.TypeName]): String = {
-    def escapePercent(s: String) = s.replaceAll("%", "%%")
     f.fields.foldLeft(escapePercent(f.prefix))((a, s) =>
       a + (s match {
         case (field, suffix) => writeField(field, tn) + escapePercent(suffix)
       })
     )
   }
+
+  def writeEventFormat(event: Event): String = {
```

Comment:
Does this function belong here? It seems event- and component-specific. Maybe the FormatCppWriter should just write formats.

Reply:
Good point -- I moved it to ComponentCppWriterUtils.